Mathlib/Mathport/Notation.lean

```diff
@@ -458,7 +458,7 @@ This command can be used in mathlib4 but it has an uncertain future and was crea
 for backward compatibility.
 -/
 elab (name := notation3) doc:(docComment)? attrs?:(Parser.Term.attributes)? attrKind:Term.attrKind
-    "notation3" prec?:(precedence)? name?:(namedName)? prio?:(namedPrio)? pp?:(prettyPrintOpt)?
+    "notation3 " prec?:(precedence)? name?:(namedName)? prio?:(namedPrio)? pp?:(prettyPrintOpt)?
```

Comment:
This isn't correct, since `precedence` shouldn't have space before it. Maybe it should be changing it so it's `pp?(ppSpace prettyPrintOpt)? items:(ppSpace notation3Item)+`?

Reply:
Thanks!

I randomly took the two `notation3` examples from the docs, added some of the optional nodes for precedence and names and pretty printed them with the various `ppSpace`s.

Staring at what is below for a while, I think that your suggestion is great!

```lean
import Mathlib.Mathport.Notation

open Lean Elab Command in
/-- shows the pretty-printed version of the following command -/
elab "#pretty " cmd:command : command => do
  logInfo (← liftCoreM do PrettyPrinter.ppCategory `command cmd)

#pretty
notation3:4 (name := hello) (priority := 50) (prettyPrint := true) "∀ᶠ " (...) " in " f ", " r:(scoped p => Filter.eventually p f) => r

-- with both `ppSpace`s outputs
notation3:4 (name := hello) (priority := 50) (prettyPrint := true) "∀ᶠ " (...) " in " f ", " r:(scoped p =>
  Filter.eventually p f) => r

-- with no `ppSpace` in front of `notation3Item` outputs
notation3:4 (name := hello) (priority := 50) (prettyPrint := true)"∀ᶠ "(...)" in "f", "r:(scoped p =>
  Filter.eventually p f) => r

-- with no `ppSpace` in front of `prettyPrintOpt` outputs
notation3:4 (name := hello) (priority := 50)(prettyPrint := true) "∀ᶠ " (...) " in " f ", " r:(scoped p =>
  Filter.eventually p f) => r

-- with no `ppSpace` at all outputs
notation3:4 (name := hello) (priority := 50)(prettyPrint := true)"∀ᶠ "(...)" in "f", "r:(scoped p =>
  Filter.eventually p f) => r

-------------------

#pretty
notation3 "MyList[" (x", "* => foldr (a b => MyList.cons a b) MyList.nil) "]" => x

-- with both `ppSpace`s outputs
notation3 "MyList[" (x ", "* => foldr (a b => MyList.cons a b) MyList.nil) "]" => x

-- with no `ppSpace` in front of `notation3Item` outputs
notation3"MyList["(x ", "* => foldr (a b => MyList.cons a b) MyList.nil)"]" => x

-- with no `ppSpace` in front of `prettyPrintOpt` outputs
notation3 "MyList[" (x ", "* => foldr (a b => MyList.cons a b) MyList.nil) "]" => x

-- with no `ppSpace` at all outputs
notation3"MyList["(x ", "* => foldr (a b => MyList.cons a b) MyList.nil)"]" => x
-/
```